### PR TITLE
T23950

### PR DESCRIPTION
**Pull request by mwleeds**

### CODE REVIEW

**Comment by mwleeds** [comment]
At the moment this exists for CI purposes. It's not necessarily CR Ready

**Comment by mwleeds** [comment]
This seems to compile and looks pretty good. CR Ready in https://phabricator.endlessm.com/T23950

**Comment by mwleeds** [comment]
Pushed a commit with the renamed function. It seems the convention of starting helper functions with `_` isn't used consistently across gnome-software.

File: plugins/eos/gs-plugin-eos.c [hunk 1]
```diff
@@ -2185,20 +2185,15 @@ should_add_os_upgrade (GsApp *os_upgrade)
...
 	return FALSE;
 }
-
-gboolean
-gs_plugin_refresh (GsPlugin *plugin,
-		   guint cache_age,
-		   GsPluginRefreshFlags flags,
-		   GCancellable *cancellable,
-		   GError **error)
+static void
```
Comment (pwithnall) [inline plugins/eos/gs-plugin-eos.c]: Nitpick: Missing blank line before this function.

File: plugins/eos/gs-plugin-eos.c [hunk 2]
```diff
@@ -2290,6 +2295,35 @@ gs_plugin_app_upgrade_download (GsPlugin *plugin,
...
+	g_autofree gchar *content_type = NULL;
+	const gchar *mimetypes_repo[] = {
+		"inode/directory",
+		NULL };
+
+	/* does this match any of the mimetypes we support */
+	content_type = gs_utils_get_content_type (file, cancellable, error);
+	if (content_type == NULL)
+		return FALSE;
+	if (g_strv_contains (mimetypes_repo, content_type)) {
```
Comment (pwithnall) [inline plugins/eos/gs-plugin-eos.c]: Nitpick: This seems like a fairly roundabout way of checking if `file` is a directory. You could use `g_file_query_file_type()` instead.

Comment (mwleeds) [reply plugins/eos/gs-plugin-eos.c]: This matches how the flatpak plugin does this, and how other plugins implement this vfunc

File: plugins/eos/gs-plugin-eos.c (hunk 1 above)
Comment (mwleeds) [reply plugins/eos/gs-plugin-eos.c]: Fixed this

**Pull request merged**